zrml/prediction-markets/src/lib.rs

```diff
@@ -639,28 +634,64 @@ mod pallet {
                 extra_weight = Self::start_subsidy(&market, market_id)?;
             }
 
-            let ids_amount: u32 = match period {
-                MarketPeriod::Block(range) => MarketIdsPerCloseBlock::<T>::try_mutate(
-                    range.end,
-                    |ids| -> Result<u32, DispatchError> {
-                        ids.try_push(market_id).map_err(|_| <Error<T>>::StorageOverflow)?;
-                        Ok(ids.len() as u32)
-                    },
-                )?,
-                MarketPeriod::Timestamp(range) => MarketIdsPerCloseTimeFrame::<T>::try_mutate(
-                    Self::calculate_time_frame_of_moment(range.end),
-                    |ids| -> Result<u32, DispatchError> {
-                        ids.try_push(market_id).map_err(|_| <Error<T>>::StorageOverflow)?;
-                        Ok(ids.len() as u32)
-                    },
-                )?,
-            };
+            let ids_amount: u32 = Self::update_auto_close(&market_id)?;
 
             Self::deposit_event(Event::MarketCreated(market_id, market_account, market));
 
             Ok(Some(T::WeightInfo::create_market(ids_amount).saturating_add(extra_weight)).into())
         }
 
+        #[pallet::weight(T::WeightInfo::edit_market(CacheSize::get()))]
+        #[transactional]
+        pub fn edit_market(
+            origin: OriginFor<T>,
+            market_id: MarketIdOf<T>,
+            oracle: T::AccountId,
+            period: MarketPeriod<T::BlockNumber, MomentOf<T>>,
+            deadlines: Deadlines<T::BlockNumber>,
+            metadata: MultiHash,
+            market_type: MarketType,
+            dispute_mechanism: MarketDisputeMechanism<T::AccountId>,
+            scoring_rule: ScoringRule,
+        ) -> DispatchResultWithPostInfo {
+            // TODO(#787): Handle Rikiddo benchmarks!
+            let sender = ensure_signed(origin)?;
+            ensure!(
+                MarketIdsForEdit::<T>::contains_key(market_id),
+                Error::<T>::MarketEditNotRequested
+            );
+            let old_market = T::MarketCommons::market(&market_id)?;
+            ensure!(old_market.creator == sender, Error::<T>::EditorNotCreator);
+            ensure!(old_market.status == MarketStatus::Proposed, Error::<T>::InvalidMarketStatus);
+
+            Self::clear_auto_close(&market_id)?;
```

Comment:
When we change `market.period`, we also need to take changes to `period.start()` into account - in other words, we need to call `clear_auto_open(&market_id)`.

Reply:
I am bit confuse, AFAIU we can deploy pool for active markets only, so not sure how auto_open storage will have anything for market which is in proposed state?

Reply:
Yes, you guys are right, `MarketIdsPerOpen*` is only modified when a pool is deployed, so we're all good. :thumbsup: